### PR TITLE
PUB-2615  | Edit Campaign, on cancel go back to the previous page

### PR DESCRIPTION
**Pull request by MayaUribe**
## Description

On Edit Campaign and click on Cancel, go back to the previous page.

## Context & Notes

When going to edit campaign, I'm saving the previous route in the router state to access when going back, if none found, redirects to the list of campaigns.

## Screenshots
Screen Recording 2020-04-03 at 03 01 PM

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/campaign-form/components/CampaignForm/index.jsx [hunk 1]
```diff
@@ -173,7 +175,7 @@ const CampaignForm = ({
             type="text"
             size="large"
             label={translations.cancel}
-            onClick={onCancelClick}
+            onClick={() => history.goBack()}
```
Comment (albennett) [inline packages/campaign-form/components/CampaignForm/index.jsx:178]: ceiling 🐈 Hey @MayaUribe I had something similar come up a while back, and wanted to mention that I don't think `history.goBack` works correctly if the user types the url in directly (unless something has changed). And I found that there wasn't a great way to detect whether `goBack` will work correctly (the history object isn't super helpful).

Comment (MayaUribe) [reply packages/campaign-form/components/CampaignForm/index.jsx:178]: Hey @albennett good one! I'll change my strategy 🙂

File: packages/campaign-form/components/CampaignForm/story.jsx [hunk 2]
```diff
@@ -19,15 +19,13 @@ storiesOf('Campaigns|CampaignForm', module)
...
       onCreateOrUpdateCampaignClick={action('saveCampaign')}
-      onCancelClick={action('cancel')}
     />
   ))
   .add('edit form', () => (
     <CampaignForm
       campaignId={campaignDetails.id}
       translations={translations.campaigns.campaignForm}
       onCreateOrUpdateCampaignClick={action('saveCampaign')}
-      onCancelClick={action('cancel')}
```
Comment (albennett) [inline packages/campaign-form/components/CampaignForm/story.jsx:30]: Curious, did you mean to remove it in the story?

File: packages/routes/index.js [hunk 3]
```diff
@@ -77,7 +77,8 @@ export const campaignCreate = {
 
 export const campaignEdit = {
   route: '/campaigns/:id/edit/',
-  goTo: ({ campaignId }) => push(`/campaigns/${campaignId}/edit`),
+  goTo: ({ campaignId, from }) =>
+    push(`/campaigns/${campaignId}/edit`, { from }),
```
Comment (albennett) [inline packages/routes/index.js:81]: ah, so the second param to push sets `from`?

File: packages/campaign-form/components/CampaignForm/story.jsx (hunk 2 above)
Comment (MayaUribe) [reply packages/campaign-form/components/CampaignForm/story.jsx:30]: That's a mistake! I'll change it back.

File: packages/routes/index.js (hunk 3 above)
Comment (MayaUribe) [reply packages/routes/index.js:81]: Yes, that's right, the second parameter for push is an optional state.

Comment (albennett) [reply packages/routes/index.js:84]: Nice solution for the direct url issue!

**Pull request merged**